common/web/keyboard-processor/tests/node/specialized-backspace.js

```diff
@@ -0,0 +1,349 @@
+
+    assert.isFalse(result.triggerKeyDefault);
+    assert.equal(contextSource.getTextBeforeCaret(), '');
+    assert.equal(contextSource.getTextAfterCaret(), '');
+    assert.isOk(result?.transcription?.transform);
+
+    const transform = result.transcription.transform;
+    assert.equal(transform.insert, '');
+    // This BKSP was to delete selected text, NOT to delete text before the caret.
+    assert.equal(transform.deleteLeft, 0);
+
+    // Is currently not what I'd expect - it reports the length of the selected string instead!
+    // Fixing that may have knock-on effects, though... and it seems to pass through
+    // the apps just fine as-is.
+    //
+    // I do have concerns for certain scripts on iOS, though... see #4538.
+    // assert.isNotOk(transform.deleteRight);
+  });
+
+  it('empty left-context, positional keyboard', () => {
+    let contextSource = new Mock('post-caret text', 0);
+    let event = new KeyEvent({
+      Lcode: Codes.keyCodes.K_BKSP,
+      Lmodifiers: 0,
+      Lstates: Codes.modifierCodes.NO_CAPS | Codes.modifierCodes.NO_NUM_LOCK | Codes.modifierCodes.NO_SCROLL_LOCK,
+      LisVirtualKey: true,
+      kName: '',
+      vkCode: Codes.keyCodes.K_BKSP,
+      device: device
+    });
+
+    const processor = new KeyboardProcessor(TEST_DEVICE, {
+      keyboardInterface: angkorWithHarness
+    });
+    const result = processor.processKeystroke(event, contextSource);
+
+    assert.isTrue(result.triggerKeyDefault);
+    assert.equal(contextSource.getTextBeforeCaret(), '');
+    assert.equal(contextSource.getTextAfterCaret(), 'post-caret text');
+    assert.isOk(result?.transcription?.transform);
+
+    const transform = result.transcription.transform;
+    assert.equal(transform.insert, '');
+    assert.equal(transform.deleteLeft, 1);
+    assert.isNotOk(transform.deleteRight);
+  });
+
+  it('empty context, mnemonic keyboard', () => {
+    let contextSource = new Mock('');
+    let event = new KeyEvent({
+      Lcode: Codes.keyCodes.K_BKSP,
+      Lmodifiers: 0,
+      Lstates: Codes.modifierCodes.NO_CAPS | Codes.modifierCodes.NO_NUM_LOCK | Codes.modifierCodes.NO_SCROLL_LOCK,
+      LisVirtualKey: true,
+      kName: '',
+      vkCode: Codes.keyCodes.K_BKSP,
+      device: device
+    });
+
+    const processor = new KeyboardProcessor(TEST_DEVICE, {
+      keyboardInterface: ipaWithHarness
+    });
+    const result = processor.processKeystroke(event, contextSource);
+
+    assert.isTrue(result.triggerKeyDefault);
+    assert.equal(contextSource.getTextBeforeCaret(), '');
+    assert.equal(contextSource.getTextAfterCaret(), '');
+    assert.isOk(result?.transcription?.transform);
+
+    const transform = result.transcription.transform;
+    assert.equal(transform.insert, '');
+    assert.equal(transform.deleteLeft, 1);
+    assert.isNotOk(transform.deleteRight);
+  });
+
+  it('final empty context, positional keyboard, rule-handled BKSP', () => {
```

Comment:
Please review and give thoughts on this test case.

At first, I was going to override any keyboard rules that took effect in this scenario... but then realized that would result in significant unpredictability should someone actually try to write such a keyboard.  Also, we can totally blame that aspect of their keyboard design - it wouldn't match a "reasonable" implementation of backspace.

I feel that this is at least very clear and consistent behavior compared to what such a keyboard would have been doing the rest of the time and thus is simpler than trying to handle various caveats that could arise from trying to override the keyboard's requested behavior instead.

But... that's my perspective on the situation.